WordPress/src/test/java/org/wordpress/android/ui/jetpack/restore/RestoreDetailsViewModelTest.kt

```diff
@@ -1,26 +1,104 @@
 package org.wordpress.android.ui.jetpack.restore
 
-import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.assertj.core.api.Assertions
 import org.junit.Before
-import org.junit.Rule
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.junit.MockitoJUnitRunner
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.TEST_DISPATCHER
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.activity.ActivityLogModel
+import org.wordpress.android.test
+import org.wordpress.android.ui.jetpack.common.JetpackListItemState.CheckboxState
+import org.wordpress.android.ui.jetpack.common.providers.JetpackAvailableItemsProvider
+import org.wordpress.android.ui.jetpack.restore.details.RestoreDetailsStateListItemBuilder
 import org.wordpress.android.ui.jetpack.restore.details.RestoreDetailsViewModel
+import org.wordpress.android.ui.jetpack.restore.details.RestoreDetailsViewModel.UiState
+import org.wordpress.android.ui.jetpack.usecases.GetActivityLogItemUseCase
+import java.util.Date
 
-@RunWith(MockitoJUnitRunner::class)
-class RestoreDetailsViewModelTest {
-    @Rule
-    @JvmField val rule = InstantTaskExecutorRule()
-
+@InternalCoroutinesApi
+class RestoreDetailsViewModelTest : BaseUnitTest() {
     private lateinit var viewModel: RestoreDetailsViewModel
+    private lateinit var availableItemsProvider: JetpackAvailableItemsProvider
+    private lateinit var stateListItemBuilder: RestoreDetailsStateListItemBuilder
+    @Mock private lateinit var getActivityLogItemUseCase: GetActivityLogItemUseCase
+    @Mock private lateinit var parentViewModel: RestoreViewModel
+    @Mock private lateinit var site: SiteModel
+
+    private val activityId = "1"
 
     @Before
-    fun setUp() {
-        viewModel = RestoreDetailsViewModel()
+    fun setUp() = test {
+        availableItemsProvider = JetpackAvailableItemsProvider()
+        stateListItemBuilder = RestoreDetailsStateListItemBuilder()
+        viewModel = RestoreDetailsViewModel(
+                availableItemsProvider,
+                getActivityLogItemUseCase,
+                stateListItemBuilder,
+                TEST_DISPATCHER
+        )
+        whenever(getActivityLogItemUseCase.get(anyOrNull())).thenReturn(fakeActivityLogModel)
     }
 
     @Test
-    fun `sample test`() {
-    } // TODO:
+    fun `when available items are fetched, the content view is shown`() = test {
```

Comment:
Minor (🔍): There is probably a type on the `then` (instead of `the`).

Reply:
in 65b23f3